src/rpc/budget.cpp

```diff
@@ -48,11 +53,110 @@ void budgetToJSON(CBudgetProposal* pbudgetProposal, UniValue& bObj)
     bObj.push_back(Pair("fValid", pbudgetProposal->fValid));
 }
 
-UniValue preparebudget(const UniValue& params, bool fHelp)
-{
-    int nBlockMin = 0;
+std::string validateURL(std::string strURL) {
+
+    // Check URL size
+    if (strURL.size() > 64)
+        return std::string("Invalid url, limit of 64 characters.");
+
+    std::vector<std::string> reqPre;
+    std::vector<std::string> reqStr;
+    std::vector<std::string> banStr;
+
+    // +++ Begin Configure URL Requirements Section
+
+    // Required initial strings; URL must contain one
+    reqPre.push_back("h");
+#ifndef ALLOW_TOLOWER
+    reqPre.push_back("H");
+#endif
+
+    // Required strings; URL must contain one
+    reqStr.push_back("ttp://");
+    reqStr.push_back("ttps://");
+#ifndef ALLOW_TOLOWER
+    reqStr.push_back("TTP://");
+    reqStr.push_back("TTPS://");
+#endif
+
+    // Banned strings; URL must not contain any
+    // banStr.push_back("example");
+#ifndef ALLOW_TOLOWER
+    // banStr.push_back("Example");
+#endif
+
+#ifdef ALLOW_TOLOWER
+   boost::algorithm::to_lower(strURL);
+#endif
+
+    // --- End Configure URL Requirements Section
+
+    // check fronts
+    bool found = false;
+    for (int i=0; i < reqPre.size() && !found; i++) {
+        if (strURL.find(reqPre[i]) == 0) found = true;
+    }
+    if (!found)
+        return std::string("Invalid URL, check scheme (e.g. https://)");
+
+    // check contains
+    found = false;
+    for (int i=0; i < reqStr.size() && !found; i++) {
+        if (strURL.find(reqStr[i]) != std::string::npos) found = true;
```

Comment:
I was going to address this in the writeup, but I had a few distractions while writing and it slipped my mind.  Yes; that's correct, and primarily the reason I put this in as a separate PR rather than updating the other.    It is a little step backwards to make way for much more expandability (e.g. if proposals become a vehicle for abuse, you can add a restriction to require them to be on ://forum.pivx.org).

From a rules standpoint; to be sure to restrict the scheme in it's entirety... the reqPre rules can be used to list out specifically from the starting position.  I would recommend it be:

```
    // Required initial strings; URL must contain one
    reqPre.push_back("http://");
    reqPre.push_back("https://");
#ifndef ALLOW_TOLOWER
    reqPre.push_back("Http://");
    reqPre.push_back("Https://");
    reqPre.push_back("HTTP://");
    reqPre.push_back("HTTPS://");
#endif
```

And not worry about hTTP ones (which would be unlikely).

Although I just realized I need another change; since it won't allow an empty list on the first two; so I need to conditionalize `if (!found && /*reqPre|reqStr*/.size() > 0)`

Reply:
discussion to continue in #950